module/shared-kernel/src/Domain/AbstractCode.php

```diff
@@ -35,12 +34,11 @@ public function __construct(string $value)
         $this->value = $value;
     }
 
-
     public static function isValid(string $value): bool
```

Comment:
Actually, it's a bit tricky - I think you should trim the `$value` as well before the validation. Otherwise, you can say it's valid via this check and then throw the exception on creating the value since it's trimmed.